### PR TITLE
Replace "sleep" sample throughout Istio docs

### PR DESCRIPTION
**Pull request by craigbox**
Having introduced a sample service named `curl` in https://github.com/istio/istio/pull/53381, this PR attempts a giant replacement of `sleep` throughout the Istio docs.

### CODE REVIEW

**Comment by craigbox** [comment]
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/15767/doc.test.dualstack_istio.io/1843579875621867520/artifacts/tests-setup-dualstack-a9b80b2bb/TestDocs/setup/additional-setup/dual-stack/test.sh/test.sh/_test_context/test.sh_output.txt

> error: the path "samples/curl/curl.yaml" does not exist

(This PR needs to wait until the automator has caught up, which I think @bleggett is working on.)

File: content/en/boilerplates/before-you-begin-egress.md [hunk 1]
```diff
@@ -9,19 +9,19 @@
...
-    $ kubectl apply -f @samples/sleep/sleep.yaml@
+    $ kubectl apply -f @samples/curl/curl.yaml@
     {{< /text >}}
 
-    Otherwise, manually inject the sidecar before deploying the `sleep` application with the following command:
+    Otherwise, manually inject the sidecar before deploying the `curl` application with the following command:
 
     {{< text bash >}}
-    $ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@)
+    $ kubectl apply -f <(istioctl kube-inject -f @samples/curl/curl.yaml@)
```
Comment (howardjohn) [inline content/en/boilerplates/before-you-begin-egress.md:24]: way off topic from this, but I thought we removed all kube-inject at some point?

**Review by howardjohn — APPROVED** [review]

**Comment by bleggett** [comment]
> https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/15767/doc.test.dualstack_istio.io/1843579875621867520/artifacts/tests-setup-dualstack-a9b80b2bb/TestDocs/setup/additional-setup/dual-stack/test.sh/test.sh/_test_context/test.sh_output.txt
> 
> > error: the path "samples/curl/curl.yaml" does not exist
> 
> (This PR needs to wait until the automator has caught up, which I think @bleggett is working on.)

Manual rebase of the automator PR from ~3 days ago here: https://github.com/istio/istio.io/pull/15768

**Comment by craigbox** [comment]
@dhawton tests pass - hopefully the commits I have somehow picked during rebase are appropriate!

File: content/en/boilerplates/before-you-begin-egress.md (hunk 1 above)
Comment (craigbox) [reply content/en/boilerplates/before-you-begin-egress.md:24]: https://github.com/istio/istio.io/issues/15777

**Review by dhawton — APPROVED** [review]

**Comment by craigbox** [comment]
Sort order bug:

```
VERIFY FAILED snip_deploy_a_sample_application_3 (timeout after 120s):
received:
"NAME                            READY   STATUS    RESTARTS   AGE
curl-6f9bc47956-5w47h           2/2     Running   0          2m1s
helloworld-v1-bfb69958d-mvwxd   2/2     Running   0          2m1s"
expected:
"NAME                             READY   STATUS    RESTARTS   AGE
helloworld-v1-776f57d5f6-s7zfc   2/2     Running   0          10s
curl-64d7d56698-wqjnm            2/2     Running   0          9s"
```

**Comment by craigbox** [comment]
#15505 flaaaaaake

/test doc.test.profile-default

**Comment by craigbox** [comment]
#15505 flaaaaaake

/test doc.test.profile-default

**Comment by craigbox** [comment]
#15505 🤬

/test doc.test.profile-default

**Pull request merged**